File: src/main/java/org/jdc/template/ui/menu/CommonMenu.java
package org.jdc.template.ui.menu;

import android.content.Context;
import android.view.MenuItem;

import org.jdc.template.InternalIntents;
import org.jdc.template.R;

import javax.inject.Inject;
import javax.inject.Singleton;

import timber.log.Timber;

@Singleton
public class CommonMenu {
    private final InternalIntents internalIntents;

    @Inject
    public CommonMenu(InternalIntents internalIntents) {
        this.internalIntents = internalIntents;
    }

    public boolean onOptionsItemSelected(Context context, MenuItem item) {
        switch (item.getItemId()) {
            case R.id.menu_item_search:
                return true;
            case R.id.menu_item_settings:
                internalIntents.showSettings(context);
                return true;
            case R.id.menu_item_about:
                internalIntents.showHelp(context);
                return true;
            default:
                Timber.i("Unknown common menu item id: %d, ignoring", item.getItemId());
                return false;
        }
    }
}

File: app/src/main/kotlin/org/jdc/template/ui/menu/CommonMenu.kt
package org.jdc.template.ui.menu

import android.view.MenuItem
import androidx.navigation.NavController
import dagger.hilt.android.scopes.FragmentScoped
import org.jdc.template.R
import org.jdc.template.ux.about.AboutRoute
import org.jdc.template.ux.settings.SettingsRoute
import timber.log.Timber
import javax.inject.Inject

@FragmentScoped
class CommonMenu
@Inject constructor() {
    fun onOptionsItemSelected(navController: NavController, item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.menu_item_search -> true
            R.id.menu_item_settings -> {
                navController.navigate(SettingsRoute.createRoute())
                true
            }
            R.id.menu_item_about -> {
                navController.navigate(AboutRoute.createRoute())
                true
            }
            else -> {
                Timber.i("Unknown common menu item id: ${item.itemId}, ignoring")
                false
            }
        }
    }
}
